portal-services/src/main/java/com/telecominfraproject/wlan/portal/controller/client/ClientPortalController.java

```diff
@@ -173,16 +138,16 @@ public Client updateClient(@RequestBody Client client) {
 
         if (paginationContext.isLastPage() || clientSessionContext.isLastPage()) {
             // no more pages available according to the context or child's context
-            LOG.debug("No more pages available when looking up Clients for customer {} equipment {} with last returned page number {}",
-                    customerId, equipmentIds, paginationContext.getLastReturnedPageNumber());
+            LOG.debug("No more pages available when looking up Clients for customer {} equipment {} macSubstring {} with last returned page number {}",
+                    customerId, equipmentIds, macSubstring, paginationContext.getLastReturnedPageNumber());
             clientResponse.setContext(paginationContext);
             return clientResponse;
         }
         
-        clientResponse = processClientResponseWithContextChildren(customerId, equipmentIds, sortBy, paginationContext);
+        clientResponse = processClientResponseWithContextChildren(customerId, equipmentIds, macSubstring, sortBy, paginationContext);
```

Comment:
something does not seem right in here. all other changes refer to getting client by mac substring, but here we are not using it at all - we're using client session by mac substring.

Reply:
maybe the existing logic (with clientServiceInterface.getSessionsForCustomer and  clientServiceInterface.get(customerId, macSet) ) needs to be used when equipmentIds are provided, but when they are not provided we can use clientServiceInterface.getForCustomer(customerId, macSubstring) by itself

Reply:
Yes it is sort of a 2 part work here. 
1. The Client API forCustomer here uses the Client Session API forCustomer to leverage the filters where the actual data is stored (e.g. equipmentId). From a portal endpoint perspective, this is how the filters are applied. 
2. The internal Client API forCustomer is not exposed to any portal endpoint, and not currently used. This is where I've merged the Client API searchByMac functionality.

Reply:
Is there a case for a user that filters for macSubstring first, then decides to add equipmentIds for further filtering? 

The equipmentids filter uses the context children to maintain its position/where it last returned. I'm not sure if it will work if the user starts with macSubstring (no context children), then adds equipmentIds to the mix (where it requests a context child)

Reply:
main thing about pagination is : parameters cannot be changed after the initial call for the first page, so we're safe there.